azurelinuxagent/common/utils/restutil.py

```diff
@@ -172,6 +172,19 @@ def _parse_url(url):
         secure = True
     return o.hostname, o.port, secure, rel_uri
 
+def _trim_url_parameters(url):
```

Comment:
Since this is a util function, it doesn't need to be a protected function. Can rename it to `trim_url_parameters` instead